Added Desktop / Rich Notifications

goibon opened the pull request:
As requested in #8 , there are now desktop notifications using the [Rich Notifications API](https://developer.chrome.com/apps/richNotifications).
If the user clicks the body of one of these then it will go to that message on Reddit to mark is as read and remove it from the list.
If the user clicks the `X` button on one, then the notification is simply hidden and the message remains unread.

goibon committed:
Added ´notifications´ to permissions in manifest.

goibon committed:
Added rich notifications.

goibon committed:
Removed console logs.

bgrawi commented:
Hi, thanks for the Pull Request! Looks pretty good so far, but I noticed while testing that it does not bring you to the right page when clicking on comment replies. It simply brings you to this for example: `https://www.reddit.com/message/messages/cvnd53r` which only works for pm's. Take a look at how the link is formatted in the regular notification window for help.

goibon committed:
Desktop notifications now takes you to the proper URL if the message was a reply to a thread or comment.

goibon committed:
Desktop notifications are now also cleared when clicking the regular notifications.

goibon committed:
Added an option to toggle desktop notifications on and off (Off by default).
Resized the box in the options page to accomodate the new option.

goibon commented:
@bgrawi You were absolutely right thank you for noticing. It should direct the user to the proper location now. I also fixed the two notification lists not properly clearing each other.
Finally I have added an option to enable / disable the new desktop notifications (Don't know why I didn't do that in the first place) and they are toggled off by default so existing users don't get an (perhaps) unpleasant suprise the next time they update :smile:

bgrawi commented:
Ok, yeah, looks good. Sorry about the delay. I've merged and published these changes to the store. Thanks for the PR!
